Bottom: fc017e2c8dc453fcd9ae68b66db4643e0acdf236
Top:    3ff6770d7c8a78981340d2d9970980ac0eb081d8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-29 20:18:32 +0100

hid/gtk: Refactor to remove some duplicated code in the layer selector


---

diff --git a/src/hid/gtk/ghid-layer-selector.c b/src/hid/gtk/ghid-layer-selector.c
index 72ec44d..248e2d3 100644
--- a/src/hid/gtk/ghid-layer-selector.c
+++ b/src/hid/gtk/ghid-layer-selector.c
@@ -477,14 +477,15 @@ ghid_layer_selector_add_layer (GHidLayerSelector *ls,
                             &iter, USER_ID_COL, &read_id,
                             SEPARATOR_COL, &is_sep,
                             ACTIVATABLE_COL, &active, -1);
-        if (!is_sep)
+
+        if (is_sep)
+          continue;
+
+        last_activatable = active;
+        if (read_id == user_id)
           {
-            last_activatable = active;
-            if(read_id == user_id)
-              {
-                new_iter = FALSE;
-                break;
-              }
+            new_iter = FALSE;
+            break;
           }
       }
     while (gtk_tree_model_iter_next (GTK_TREE_MODEL (ls->list_store), &iter));
@@ -501,18 +502,7 @@ ghid_layer_selector_add_layer (GHidLayerSelector *ls,
                               SEPARATOR_COL, TRUE, -1);
         }
       /* Create new layer */
-      new_layer = malloc (sizeof (*new_layer));
       gtk_list_store_append (ls->list_store, &iter);
-      gtk_list_store_set (ls->list_store, &iter,
-                          STRUCT_COL, new_layer,
-                          USER_ID_COL, user_id,
-                          VISIBLE_COL, visible,
-                          COLOR_COL, color_string,
-                          TEXT_COL, name,
-                          FONT_COL, activatable ? NULL : "Italic",
-                          ACTIVATABLE_COL, activatable,
-                          SEPARATOR_COL, FALSE,
-                          -1);
     }
   else
     {
@@ -521,18 +511,21 @@ ghid_layer_selector_add_layer (GHidLayerSelector *ls,
       gtk_tree_model_get (GTK_TREE_MODEL (ls->list_store), &iter,
                           STRUCT_COL, &new_layer, -1);
       free_ldata (ls, new_layer);
-      new_layer = malloc (sizeof (*new_layer));
-
-      gtk_list_store_set (ls->list_store, &iter,
-                          STRUCT_COL, new_layer,
-                          VISIBLE_COL, visible,
-                          COLOR_COL, color_string,
-                          TEXT_COL, name,
-                          FONT_COL, activatable ? NULL : "Italic",
-                          ACTIVATABLE_COL, activatable,
-                          -1);
     }
 
+  new_layer = malloc (sizeof (*new_layer));
+
+  gtk_list_store_set (ls->list_store, &iter,
+                      STRUCT_COL, new_layer,
+                      USER_ID_COL, user_id,
+                      VISIBLE_COL, visible,
+                      COLOR_COL, color_string,
+                      TEXT_COL, name,
+                      FONT_COL, activatable ? NULL : "Italic",
+                      ACTIVATABLE_COL, activatable,
+                      SEPARATOR_COL, FALSE,
+                      -1);
+
   /* -- Setup new actions -- */
   vname = g_strdup_printf ("LayerView%d", ls->n_actions);
   pname = g_strdup_printf ("LayerPick%d", ls->n_actions);
